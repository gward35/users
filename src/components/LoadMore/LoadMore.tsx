import { Button } from '@chakra-ui/react'

export type LoadMoreProps = {
  isDisabled: boolean
  isLoading: boolean
  onClick: () => void
}

export const LoadMore = ({ isDisabled, isLoading, onClick }: LoadMoreProps) => {
  return (
    <Button
      bgColor="primary.100"
      color="white"
      data-testid="loadMore"
      _hover={{
        bgColor: 'primary.200',
      }}
      isDisabled={isDisabled}
      isLoading={isLoading}
      mx={4}
      mb={8}
      onClick={onClick}
    >
      Load More
    </Button>
  )
}

export default LoadMore
