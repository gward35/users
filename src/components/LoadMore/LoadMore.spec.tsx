import { screen, render, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import LoadMore from './LoadMore'
const user = userEvent.setup()

describe('Load More', () => {
  const PROPS = {
    onClick: jest.fn(),
    isLoading: false,
  }

  it('loads component', () => {
    const utils = render(<LoadMore {...PROPS} />)
    expect(utils.container).toMatchSnapshot()
    expect(screen.getByTestId('loadMore')).toBeInTheDocument()
  })

  it('clicks', async () => {
    render(<LoadMore {...PROPS} />)
    user.click(screen.getByTestId('loadMore'))
    await waitFor(() => {
      expect(PROPS.onClick)
    })
  })
})
