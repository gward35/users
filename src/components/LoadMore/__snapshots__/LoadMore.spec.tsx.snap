// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load More loads component 1`] = `
<div>
  <button
    class="chakra-button css-vkxzif"
    data-testid="loadMore"
    type="button"
  >
    Load More
  </button>
</div>
`;
